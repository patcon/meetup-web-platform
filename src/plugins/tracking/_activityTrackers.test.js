import { getLogger } from './activity';
import { getTrackActivity, getTrackApiResponses } from './_activityTrackers';
jest.mock('uuid', () => {
	return {
		v4: () => 'test-v4-uuid',
	};
});

describe('getTrackActivity', () => {
	it('records the expected object', () => {
		const logger = getLogger('MUP_WEB');
		const trackOpts = {
			memberCookieName: 'member',
			log: logger,
		};
		const memberId = 1337;
		const request = {
			id: 'test-request-id-1234',
			url: {
				pathname: '/mu_api',
			},
			headers: {
				'user-agent': 'test user agent',
			},
			method: 'get',
			payload: '',
			query: {},
			info: {
				referrer: 'http://example.com/referrer',
			},
			plugins: {
				tracking: {},
			},
			state: {
				member: `id=${memberId}&status=1`,
			},
		};
		request.trackApiResponses = getTrackApiResponses(trackOpts)(request);
		const trackActivity = getTrackActivity(trackOpts);
		const trackRequestActivity = trackActivity(request);
		const queryResponses = [];
		expect(trackRequestActivity(queryResponses)).toMatchSnapshot();
	});
});
