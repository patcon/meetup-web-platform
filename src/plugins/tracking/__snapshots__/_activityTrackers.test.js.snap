// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getTrackActivity records the expected object 1`] = `
Object {
  "agent": "test user agent",
  "apiRequests": Array [],
  "browserId": "test-v4-uuid",
  "description": "nav",
  "ip": "",
  "isUserActivity": true,
  "memberId": 1337,
  "mobileWeb": false,
  "platform": "WEB",
  "platformAgent": "MUP_WEB",
  "referer": "/referrer",
  "requestId": "test-request-id-1234",
  "timestamp": "2017-08-31T17:53:40.537Z",
  "trackId": "test-v4-uuid",
  "trax": Object {},
  "url": "/mu_api",
}
`;
